src/spa_solver_visualizer.cpp

```diff
@@ -0,0 +1,103 @@
+#include "spa_solver_visualizer.h"
+#include "ros/console.h"
+#include <pluginlib/class_list_macros.h>
+
+namespace karto_plugins{
+
+SPASolverVisualizer::SPASolverVisualizer():marker_count_(0),visualizer_initialized_(false)
+{
+}
+
+void SPASolverVisualizer::initialize(const boost::shared_ptr<karto::ScanSolver>& solver){
```

Comment:
@stwirth this is a WIP, I haven't corrected the formatting everywhere yet. 
I tried to follow the idea of creating another plugin interface for the visualizers : karto::ScanSolverVisualizer (names can be improved), following the model of karto::ScanSolver.
This creates a little OOD complication since visualizers will work only with specific solvers, but there is no inheritance relationship between solvers and visualizers. (I may be missing a design pattern here)
So a visualizer like SPASolverVisualizer needs to store a pointer to a SPASolver object, but we can only pass a ptr to a generic ScanSolverVisualizer through the initialize() function called in the slam_karto node.
That means I have to do a downcast - here with boost::dynamic_pointer_cast - which does not look very pretty to me for error-handling.
Maybe you had something easier in mind when you suggested another plugin interface for visualizers ?